consensus/tendermint/tendermint.go

```diff
@@ -273,3 +233,24 @@ func (t *stateMachine[V, H, A]) findProposal(r Round) *CachedProposal[V, H, A] {
 		ID:       utils.HeapPtr((*v.Value).Hash()),
 	}
 }
+
+func (t *stateMachine[V, H, A]) onTimeoutPropose(h types.Height, r types.Round) Action[V, H, A] {
+	if t.state.height == h && t.state.round == r && t.state.step == types.StepPropose {
+		return t.setStepAndSendPrevote(nil)
+	}
+	return nil
+}
+
+func (t *stateMachine[V, H, A]) onTimeoutPrevote(h types.Height, r types.Round) Action[V, H, A] {
+	if t.state.height == h && t.state.round == r && t.state.step == types.StepPrevote {
+		return t.setStepAndSendPrecommit(nil)
+	}
+	return nil
+}
+
+func (t *stateMachine[V, H, A]) onTimeoutPrecommit(h types.Height, r types.Round) Action[V, H, A] {
+	if t.state.height == h && t.state.round == r {
+		return t.startRound(r + 1)
+	}
+	return nil
+}
```

Comment:
I think these lines should still be in a separate `timeout.go` file. Currently `tendermint.go` only contains mostly utils functions.